src/plugins/autocompletion_providers/Git.ts

```diff
@@ -84,181 +141,310 @@ const notStagedFiles = unique(async(context: AutocompletionContext): Promise<Sug
         description: "Move or rename a file, a directory, or a symlink.",
+        aliases: [],
     },
     {
         name: "notes",
         description: "Add or inspect object notes.",
+        aliases: [],
     },
     {
         name: "pull",
         description: "Fetch from and integrate with another repository or a local branch.",
+        provider: combine([
+            longFlag("rebase"),
+            longFlag("no-rebase"),
+            commonMergeOptions,
+            fetchOptions,
+        ]),
+        aliases: [],
     },
     {
         name: "push",
         description: "Update remote refs along with associated objects.",
+        provider: pushOptions,
+        aliases: [],
     },
     {
         name: "rebase",
         description: "Forward-port local commits to the updated upstream head.",
+        aliases: [],
     },
     {
         name: "reset",
         description: "Reset current HEAD to the specified state.",
+        aliases: [],
     },
     {
         name: "revert",
         description: "Revert some existing commits.",
+        aliases: [],
     },
     {
         name: "rm",
         description: "Remove files from the working tree and from the index.",
+        aliases: [],
     },
     {
         name: "shortlog",
         description: "Summarize git log output.",
+        aliases: [],
     },
     {
         name: "show",
         description: "Show various types of objects.",
+        aliases: [],
     },
     {
         name: "stash",
         description: "Stash the changes in a dirty working directory away.",
+        aliases: [],
     },
     {
         name: "status",
         description: "Show the working tree status.",
+        provider: statusOptions,
+        aliases: [],
     },
     {
         name: "submodule",
         description: "Initialize, update or inspect submodules.",
+        aliases: [],
     },
     {
         name: "tag",
         description: "Create, list, delete or verify a tag object signed with GPG.",
+        aliases: [],
     },
     {
         name: "worktree",
         description: "Manage multiple worktrees.",
+        aliases: [],
     },
-].map(subCommand => new Suggestion().withValue(subCommand.name).withDescription(subCommand.description).withStyle(styles.command).withSpace());
+];
+
+async function loadAllSubcommands() {
+    const text = await executeCommand("git", ["help", "-a"], process.env.HOME);
+    const matches = text.match(/  ([\-a-zA-Z0-9]+)/gm);
+
+    if (matches) {
+        matches
+            .filter((match) => match.indexOf("--") === -1)
+            .forEach(async(match) => {
+                const name = match.trim();
+                const data = find(subCommandsData, {name});
+
+                if (!data) {
+                    subCommandsData.push({
+                        name,
+                        description: "",
+                        aliases: [],
+                    });
+                }
+            });
+    }
+}
```

Comment:
Oh, man. In place modification is so not cool. I'd much rather have `combine([staticSubcommands, dynamicSubcommands])` with some performance penalty that this.